File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& p, int l) {
        int i=0,j=p.size()-1,ans=0,temp;
        sort(p.begin(),p.end());
        while(i<=j)
        {
            if(p[i]+p[j]<=l)
               ans++,i++,j--;
           else ans++,j--;
        }
        return ans;
    }
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution(object):
    def numRescueBoats(self, people, limit):
        people.sort()
        i=0
        j=len(people)-1
        ans=0
        while(i<=j):
            if(people[i]+people[j]<=limit):
                ans+=1
                i+=1
                j-=1
            else:
                ans+=1
                j-=1
        return ans
        
        